sdk/maps/maps-search/src/mapsSearchClient.ts

```diff
@@ -0,0 +1,843 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AzureKeyCredential, TokenCredential, isTokenCredential } from "@azure/core-auth";
+import {
+  GeoJsonCircleOrPolygonFeature,
+  GeoJsonCircleOrPolygonFeatureCollection,
+  GeoJsonFeatureCollection,
+  GeoJsonLineString,
+  GeoJsonPolygonFeature,
+  LatLon,
+  SearchGeometry,
+  StructuredAddress,
+} from "./models/models";
+import {
+  BatchPoller,
+  BatchPollerProxy,
+  createAzureMapsKeyCredentialPolicy,
+  createMapsClientIdPolicy,
+} from "../../maps-common/src";
```

Comment:
Shouldn't this be imported from `@azure/maps-common`?

Reply:
We recommend publishing `@azure/maps-common`. Using a local reference has brought us more problems than it is worth. If we publish the common package any major changes to it can be versioned which is a big advantage.

@timovv - fyi as he's familiar with the problems we've faced with sharing code this way.

Reply:
I see. I agree that the versioned package should bring benefits to us. Let me change this.

@timovv If you don't mind, could you share your experience of the problem you've encountered? 🤔

Reply:
@joheredi Now the `@azure/maps-common` has been changed to a published package, please take a look. Thank you.

Reply:
The main issue with a local reference is that making a breaking change to the local code means you must immediately change all the packages that depended on it. We had to do this with the Key Vault SDK a few months back, and it resulted in a giant PR which was very difficult to review. We ran into a couple of issues that we had to hotfix as a result of that which would have most likely been caught otherwise. Publishing means you can roll out those changes incrementally across your packages, which is a lot safer!

Reply:
@timovv Understood. Thanks for the detailed explanation 👍🏼